# TESTING OF HYPOTHESIS

The question is:
1. Carry out descriptive analytics on the dataset.
2. Check whether female and male students scored the same marks.
3. Whether test preparation helps the students

The given dataset, named, "Week_8_Q&A_dataset - Sheet1", consists of 28 rows and 3 columns namely Gender, Test preparation and Total Marks. 

# 1. Descriptive analytics on the dataset

In [1]:
import numpy as np 
import pandas as pd
import seaborn as sns
import scipy.stats as stats

In [2]:
data=pd.read_csv("Week_8_Q&A_dataset - Sheet1.csv")

In [12]:
data.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [7]:
data.shape

(28, 3)

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


In [9]:
data.isna().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

Hence, no null values present in the dataset.

In [22]:
data.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

# 2. Whether female and male students scored the same marks

In [13]:
male = data.query('Gender == "male"')['Total Marks']
female = data.query('Gender == "female"')['Total Marks']

In [14]:
male

0     14
6     30
8     18
9     24
10    44
11    23
12    43
13    28
18    47
19    36
20    14
21    44
Name: Total Marks, dtype: int64

In [15]:
female

1     28
2     18
3     48
4     21
5     40
7     12
14    41
15    13
16    48
17    45
22    41
23    27
24    27
25    38
26    50
27    43
Name: Total Marks, dtype: int64

We have to check whether male and female score the same mark. For that we should carry out two-tailed t-test such that

NULL HYPOTHESIS: Average mark of male and female are same against

ALTERNATIVE HYPOTHESIS:Average mark of male and female are not the same 
    
Next, consider the p-value,

If the p-value is less than your significance level, you can reject the null hypothesis and conclude that the effect is statistically significant. In other words, the evidence in your sample is strong enough to be able to reject the null hypothesis at the population level.

Here, the significane level is assumed to be 0.05.

In [23]:
t_value,p_value = stats.ttest_ind(male, female, equal_var=True)
t_value,p_value

(-0.6940885268848668, 0.4937822317960763)

INSIGHT:
Here, the p-value, 0.4937822317960763, is more than the significance level, 0.05.

Hence, we should accept the null hypothesis.

So, we can conclude that the male and female scored the same mark.

# 3. Whether test preparation helps the students

In [24]:
test_prep = data[data["Test preparation"]=="completed"]["Total Marks"]
no_test_prep = data[data["Test preparation"]=="none"]["Total Marks"]

In [25]:
test_prep

5     40
10    44
12    43
16    48
21    44
26    50
27    43
Name: Total Marks, dtype: int64

In [26]:
no_test_prep

0     14
1     28
2     18
3     48
4     21
6     30
7     12
8     18
9     24
11    23
13    28
14    41
15    13
17    45
18    47
19    36
20    14
22    41
23    27
24    27
25    38
Name: Total Marks, dtype: int64

We have to check whether test preparation helps the students. i.e; to check is there any increase in average mark of students those who completed the test preparation. For that we should carry out one-tailed t-test such that

NULL HYPOTHESIS: Average mark of students those who completed test prepartion and those of not completed are same against

ALTERNATIVE HYPOTHESIS:Average mark of students those who completed test prepartion is greater than those of not completed

In [27]:
t_value,p_value = stats.ttest_ind(test_prep, no_test_prep, equal_var=True)
t_value,p_value

(3.6143637527769217, 0.0012671292457320106)

INSIGHT:
Here, the p-value, 0.0012671292457320106, is less than the significance level, 0.05.

Hence, we should reject the null hypothesis.

So, we can conclude that the average mark of students those who completed test prepartion is greater than those of not completed.

Hence the test prepartion helps the students.